packages/forma-36-react-components/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -121,6 +87,40 @@ export const Autocomplete = <T extends {}>({
   const listRef = useRef<HTMLDivElement | null>(null);
   const inputRef = useRef<HTMLInputElement | null>(null);
 
+  const initialState: State = {
+    isOpen: dropdownProps?.isOpen ?? false,
+    query: '',
+    highlightedItemIndex: null,
+  };
+
+  const reducer = (state: State, action: Action): State => {
```

Comment:
I'm not sure about moving the reducer inside the component. I'd rather change ITEM_SELECTED action and explicitly pass state that needs to be assigned (initialState in this case)

Reply:
Good catch, I agree 👍